### ISSUE TITLE
false positives for SC2086 - Double quote to prevent globbing and word splitting

### ISSUE THREAD

**Issue by subodhdharma** [hadolint/hadolint#791]
- [x] This is a bug report
- [ ] This is a feature request
- [x] I searched existing issues before opening this one

### Expected behavior
hadolint should not report warning on the following Dockerfile lines

```shell
FROM ${GO_DOCKER_BASE} AS gobase
RUN apt-get install -y --no-install-recommends git \
	&& apt-get cache clean \
	&& rm -rf /var/lib/apt/lists/*
```
### Actual behavior
hadolint reports SC2086 - info: Double quote to prevent globbing and word splitting.

### Steps to reproduce the behavior

- use the above line to compose a Dockerfile
- run hadolint tools on the Dockerfile.

```bash
 hadolint version -- 2.8.0
 SC2086 info: Double quote to prevent globbing and word splitting.
```

**Dockerfile** (if relevant)

```Dockerfile
FROM ${GO_DOCKER_BASE} AS gobase
RUN apt-get install -y --no-install-recommends git \
	&& apt-get cache clean \
	&& rm -rf /var/lib/apt/lists/*
```

**Additional environment details (OS, stack version, etc.)**

**Comment by wedsa5** [comment]
To add, I also got the warning for this and my docker build actually fails with double quotes. So I am forced to ignore this hadolint warning.

```Dockerfile
RUN update-ca-trust && \
    keytool \
      -import \
      -file /etc/pki/ca-trust/source/anchors/redacted.crt \
      -alias redacted \
      -cacerts \
      -noprompt \
      -storepass changeit
```

**Comment by mithunkrb** [comment]
I also encountered the same problem and was able to resolve as below:
Problematic Dockerfile:
ARG KONG_VERSION="3.5.0"
FROM kong:${KONG_VERSION}-ubuntu as builder

Corrected Dockerfile:
ARG KONG_VERSION="3.5.0"
FROM kong:"${KONG_VERSION}"-ubuntu as builder
# Note the double quote around ${KONG_VERSION} in FROM line
